components/eamxx/src/share/io/scorpio_output.cpp

```diff
@@ -1038,11 +1038,42 @@ register_variables(const std::string& filename,
   }
   // Now register the average count variables
   if (m_track_avg_cnt) {
+    std::string unitless = "unitless";
     for (const auto& name : m_avg_cnt_names) {
       const auto layout = m_layouts.at(name);
       auto vec_of_dims   = set_vec_of_dims(layout);
-      scorpio::define_var(filename, name, "unitless", vec_of_dims,
-                          "real",fp_precision, m_add_time_dim);
+      if (mode==scorpio::FileMode::Append) {
```

Comment:
This is essentially a copy/paste of the same check used above for `m_field_names`